src/argilla/server/apis/v1/handlers/fields.py

```diff
@@ -47,12 +52,18 @@ async def update_field(
 ):
     field = await _get_field(db, field_id)
 
+    if field_update.settings and field_update.settings.type != field.settings["type"]:
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=f"Field type cannot be changed. Expected '{field.settings['type']}' but got '{field_update.settings.type}'",
+        )
+
     await authorize(current_user, FieldPolicyV1.update(field))
```

Comment:
I would say to check action grants for the connected user first, and then apply action validations. Otherwise, we may share the sensible info with unauthorized users
```suggestion
    await authorize(current_user, FieldPolicyV1.update(field))
    
    if field_update.settings and field_update.settings.type != field.settings["type"]:
        raise HTTPException(
            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
            detail=f"Field type cannot be changed. Expected '{field.settings['type']}' but got '{field_update.settings.type}'",
        )
```

Reply:
wops! my bad